src/messaging/ExchangeMessageDispatch.cpp

```diff
@@ -137,5 +138,37 @@ CHIP_ERROR ExchangeMessageDispatch::OnMessageReceived(uint32_t messageCounter, c
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ExchangeMessageDispatch::OnGroupMessageReceived(uint32_t messageCounter, const PayloadHeader & payloadHeader,
+                                                           const Transport::PeerAddress & peerAddress, MessageFlags msgFlags,
+                                                           ReliableMessageContext * reliableMessageContext)
+{
+    ReturnErrorCodeIf(!GroupMessagePermitted(payloadHeader.GetProtocolID().GetProtocolId(), payloadHeader.GetMessageType()),
+                      CHIP_ERROR_INVALID_ARGUMENT);
+
+    return CHIP_NO_ERROR;
+}
+
+bool ExchangeMessageDispatch::GroupMessagePermitted(uint16_t protocol, uint8_t type)
+{
+    switch (protocol)
+    {
+    case Protocols::InteractionModel::Id.GetProtocolId():
+        switch (type)
+        {
+        case static_cast<uint8_t>(Protocols::InteractionModel::MsgType::WriteRequest):
+        case static_cast<uint8_t>(Protocols::InteractionModel::MsgType::InvokeCommandRequest):
+            return true;
```

Comment:
This is layer crossing. This should be done at the IM layer in OnMessageReceived. Please remove this as it couples MessageLayer *directly* with IM, which is neither necessary nor allowed.

Reply:
I agree with the comment, but the goal of this PR was to address Boris's comments on having the flow go through `MessagePermited`. By moving these changes to the IM layer, this PR and the issue both become pointless. 
@bzbarsky-apple are you ok with keeping the code as is in the `ExchangeContext `?

A task already exist in order to add a check in IM.  #11743